### DISCUSSION TITLE
Questions on use with axios.

### DISCUSSION THREAD

**Discussion by rachitpant** [ElMassimo/js_from_routes#29] [Q&A] [answered]
1)  Is Axios being used.
I installed [@js-from-routes/axios](https://github.com/ElMassimo/js_from_routes/tree/main/packages/axios) , but when the apis are generated , I think its still using fetch.
Here are the contents of one of the api file.
```
// JsFromRoutes CacheKey 8025fd79055076a8a9153c66e74136c1
//
// DO NOT MODIFY: This file was automatically generated by JsFromRoutes.
import { definePathHelper } from '@js-from-routes/client'

export default {
  fetchUser: definePathHelper('get', '/api/fetch_user'),
  staffUsers: definePathHelper('get', '/api/staff_users'),
}
```
Can you please confirm if its using axios.

2) How to configure axios

Also we are using axios already , and have configured an interceptor like this.
```
const httpClient = axios.create(config);
// define the intereptors
httpClient.interceptors.request.use(csrfInterceptor);
httpClient.interceptors.response.use(responseInterceptor, errorInterceptor);
export default httpClient;
// import http client to build apis manually
```
I think we wont need to configure crsf iterceptor anymore , but can how can we configure the other interceptors. Maybe an example.

3. Any suggestions where we should configure the client with the config object.

**Comment by rachitpant** [comment]
I got the answer for the first and the third question by looking at the ping-crm repo.

**Comment by ElMassimo** [comment, marked as answer]
Hi Rachit, I'll answer all three in case someone runs into this.

---

> Is Axios being used?

You should [configure `client_library`](https://js-from-routes.netlify.app/config/#client-library) to use `@js-from-routes/axios` instead:

```rb
# config/initializers/js_from_routes.rb
   
if Rails.env.development?
  JsFromRoutes.config do |config|
    config.client_library = '@js-from-routes/axios'
  end
end
```

Make sure to [regenerate](https://js-from-routes.netlify.app/guide/codegen.html#code-generation-%F0%9F%A4%96) the helpers after making this change.

--- 

> How to configure axios?

You can modify the configuration by importing [`AxiosConfig`](https://github.com/ElMassimo/js_from_routes/blob/main/packages/axios/src/index.ts#L5-L10):

```js
import { AxiosConfig } from '@js-from-routes/axios'

AxiosConfig.instance = httpClient
```

---

> Any suggestions where we should configure the client with the config object?

[Read this section in the docs](https://js-from-routes.netlify.app/client/#configuring-requests-%E2%9A%99%EF%B8%8F).

It's usually convenient to configure it before any requests are fired, so importing a file that does the configuration at the start of an entrypoint would be a good idea.

**Comment by rachitpant** [comment]
Thank You.
